### PR TITLE
cortx-33358

### PR DESCRIPTION
**Pull request by apurwamohite**
Signed-off-by: ApurwaMohite <55824145+ApurwaMohite@users.noreply.github.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: tests/ha/single_pod_restart/test_server_pod_restart.py [hunk 1]
```diff
@@ -1456,3 +1456,170 @@ def test_ios_during_server_pod_restart(self):
...
+
+        LOGGER.info("Step 5.3: Verifying responses from DELETEs background process")
+        del_resp = tuple()
+        while len(del_resp) != 2:
+            del_resp = del_output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        if not del_resp:
+            assert_utils.assert_true(False, "Background process failed to do DELETEs")
+        event_del_bkt = del_resp[0]  # Contains buckets when event was set
+        fail_del_bkt = del_resp[1]  # Contains buckets which failed when event was clear
+        assert_utils.assert_false(len(fail_del_bkt) or len(event_del_bkt),
```
Comment (sonalk0209) [inline tests/ha/single_pod_restart/test_server_pod_restart.py:1608]: why all deletes will pass as we do not have any retries for deletes?

File: tests/ha/single_pod_restart/test_server_pod_restart.py [hunk 2]
```diff
@@ -1456,3 +1456,170 @@ def test_ios_during_server_pod_restart(self):
...
+        LOGGER.info("Step 2.2: Start READs and verify DI on the written data in background")
+        output_rd = Queue()
+        args = {'s3userinfo': list(users.values())[0], 'log_prefix': test_prefix_read,
+                'nclients': 1, 'nsamples': 10, 'skipwrite': True, 'skipcleanup': True,
+                'output': output_rd, 'setup_s3bench': False, 'event_set_clr': event_set_clr,
+                'max_retries': 5}
+        thread_rd = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,),
+                                     kwargs=args)
+        thread_rd.daemon = True  # Daemonize thread
+        LOGGER.info("Step 2.2: Successfully started READs and verify DI on the written data in "
```
[the remark this answers is not in the document]
Comment (apurwamohite) [reply tests/ha/single_pod_restart/test_server_pod_restart.py:1527]: Starting all threads at one place, will remove logger from here.

File: tests/ha/single_pod_restart/test_server_pod_restart.py [hunk 3]
```diff
@@ -1456,3 +1456,170 @@ def test_ios_during_server_pod_restart(self):
...
+                    "background")
+
+        LOGGER.info("Step 2.3: Starting DELETEs of %s buckets in background", del_bucket)
```
[the remark this answers is not in the document]
Comment (apurwamohite) [reply tests/ha/single_pod_restart/test_server_pod_restart.py:1530]: Yes, deleting all the buckets.

File: tests/ha/single_pod_restart/test_server_pod_restart.py (hunk 1 above)
Comment (apurwamohite) [reply tests/ha/single_pod_restart/test_server_pod_restart.py:1608]: We have retries for deletes as well. And I have added retry count.

File: tests/ha/single_pod_restart/test_server_pod_restart.py [hunk 4]
```diff
@@ -1456,3 +1456,164 @@ def test_ios_during_server_pod_restart(self):
...
+        LOGGER.info("Step 2: Start WRITEs, READs and DELETEs with variable object sizes "
+                    "during server pod restart using kubectl delete")
+        LOGGER.info("Step 2.1: Start WRITEs with variable object sizes in background")
+        test_prefix_write = f'test-44854-write-{t_t}'
+        output_wr = Queue()
+        event_set_clr = [False]
+        args = {'s3userinfo': list(users.values())[0], 'log_prefix': test_prefix_write,
+                'nclients': 2, 'nsamples': 10, 'skipread': True, 'skipcleanup': True,
+                'output': output_wr, 'setup_s3bench': False, 'event_set_clr': event_set_clr,
+                'max_retries': 5}
```
Comment (ankita-93) [inline tests/ha/single_pod_restart/test_server_pod_restart.py:1511]: Is this going to depend on some parameter. Can we add this in HA config as a common value as we are going to use it at many places.

File: libs/ha/ha_common_libs_k8s.py [hunk 5]
```diff
@@ -793,7 +793,7 @@ def restore_pod(self, pod_obj, restore_method, restore_params: dict = None,
 
     def event_s3_operation(self, event, setup_s3bench=True, log_prefix=None, s3userinfo=None,
                            skipread=False, skipwrite=False, skipcleanup=False, nsamples=10,
-                           nclients=10, output=None, event_set_clr=None,
+                           nclients=10, output=None, event_set_clr=None, max_retries=None,
```
[the remark this answers is not in the document]
Comment (apurwamohite) [reply libs/ha/ha_common_libs_k8s.py:796]: No, we don't want to retry in normal IOs, so None is the default value.

File: tests/ha/single_pod_restart/test_server_pod_restart.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sonalk0209) [reply tests/ha/single_pod_restart/test_server_pod_restart.py:1526]: why doing that ? Can we delete some and maybe afterwards check for remaining and try running read-verify on the same?

File: tests/ha/single_pod_restart/test_server_pod_restart.py (hunk 1 above)
Comment (sonalk0209) [reply tests/ha/single_pod_restart/test_server_pod_restart.py:1602]: So are we running with force option? I guess the retry is for force only right?

**Review by sonalk0209 — APPROVED** [review]

File: tests/ha/single_pod_restart/test_server_pod_restart.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (apurwamohite) [reply tests/ha/single_pod_restart/test_server_pod_restart.py:1526]: So I am deleting buckets which are created for delete operation only. On remaining buckets I am doing write/read/verify.

File: tests/ha/single_pod_restart/test_server_pod_restart.py (hunk 1 above)
Comment (apurwamohite) [reply tests/ha/single_pod_restart/test_server_pod_restart.py:1602]: Yes, we are running with force option.

File: tests/ha/single_pod_restart/test_server_pod_restart.py [hunk 6]
```diff
@@ -750,7 +750,7 @@ def test_server_pod_restart_kubectl_delete(self):
         LOGGER.info("Step 2: Start WRITEs, READs and DELETEs with variable object sizes "
                     "during server pod restart using kubectl delete")
         LOGGER.info("Step 2.1: Start WRITEs with variable object sizes in background")
-        test_prefix_write = f'test-write-34261-{t_t}'
+        test_prefix_write = f'test-34261-write-{t_t}'
```
Comment (rahul-hatwar) [inline tests/ha/single_pod_restart/test_server_pod_restart.py:753]: why change this?

**Review by rahul-hatwar — APPROVED** [review]
LGTM

File: libs/ha/ha_common_libs_k8s.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ankita-93) [reply libs/ha/ha_common_libs_k8s.py:796]: -s3MaxRetries int
        The maximum number of times that a request will be retried for failures. Defaults to -1, which defers the max retry setting to the service specific configuration (default -1)

**Review by kachhwahadivya — APPROVED** [review]

**Pull request merged**